### PR TITLE
Parser changes for new HPI website

### PR DESCRIPTION
**Pull request by gersseba**
changes to support new website
full support of utf-8 characters
simple guess for short form of course names

### CODE REVIEW

File: parser/parser.py [hunk 1]
```diff
@@ -1,268 +1,263 @@
...
-			else:
-				if (year < 9):
-					url = "http://www.hpi.uni-potsdam.de/studium/lehrangebot/lehrangebotsarchiv/lehrangebotsarchiv_ws_0" + str(year) + "0" + str(year+1) + ".html"
-				elif (year == 9):
-					url = "http://www.hpi.uni-potsdam.de/studium/lehrangebot/lehrangebotsarchiv/lehrangebotsarchiv_ws_0" + str(year) + str(year+1) + ".html"
-				else:
-					url = "http://www.hpi.uni-potsdam.de/studium/lehrangebot/lehrangebotsarchiv/lehrangebotsarchiv_ws_" + str(year) + str(year+1) + ".html"
-	print url
-	return url
+    # returns list of courses for semester
```
Comment (psieg) [inline parser/parser.py]: afair ``"""text"""`` is the equivalent to Javadoc and probably better here, even if no one is ever going to read it...

File: parser/parser.py [hunk 2]
```diff
@@ -1,268 +1,263 @@
...
-
-	for line in site:
-		if (line.strip().startswith('<td style="width: 300px; paddi')):
-			break
-
-	line = str(site.next())
-
-#	rows = line.split("<tr>")
-
-#	urls = []
```
Comment (psieg) [inline parser/parser.py]: deeaaad code ;)

**Comment by knub** [comment]
I am not sure, whether this is a good idea to use the parser. We did this at the very beginning of 180, but quickly found out, that everything needs to be hand-checked anyways. First, because the website is subject to many small changes and potential parser errors. Second, and most important: The website is not the authoritative source, the PDF documents are. The changes are not always in sync, and the data on the website is not always copied correctly.

File: parser/parser.py (hunk 2 above)
Comment (gersseba) [inline parser/parser.py]: What do you mean? The commented lines?

Comment (psieg) [inline parser/parser.py]: indeed

Comment (gersseba) [inline parser/parser.py]: But that is old code that I removed. These lines don't exist anymore.

Comment (psieg) [inline parser/parser.py]: egh. nvm

**Comment by gersseba** [comment]
I see no reason not to use it. I don't know if you have tried adding the courses manually. I have, and it's a pain in the ass, and very error prone. For this semester Aggharta did it, and there are already at least two errors (Trends in BPM Research is called "BPMN" which is something different, and also it is added for WS15/16 instead of SS16). Both of these would not have happened if a parser was used. There might be more.
So for the first point: as long as it works on the website, it is faster & safer than manual work. When it doesn't work anymore, it will have to be adapted, as it was now.
For the second point: It is sad that the website ist always correct & up to date and that is an entirely different problem that should be tackled. But as long as no one is writing a parser for the pdf, I still believe that the web parser more convinient and less faulty than the current approach. 
Also the website has more information that the pdf is lacking.

**Comment by knub** [comment]
> I don't know if you have tried adding the courses manually.

Yes. For four years now every semester ;-).

Adding the courses takes an hour of concentrated work per semester, with double- and triple-checking included. For that, I know that I can trust the data completely and do not need to double-check, when I am using the software.

> it is faster & safer than manual work.
It's definitely not safer than proper manual work, which does not take too long on a per-semester basis, as I wrote above.

> When it doesn't work anymore, it will have to be adapted, as it was now.

How will you know, that something breaks? You'd have to manually check the website every semester and see whether that corresponds to the parser output.
The website is maintained by persons, who do not care/know about machine readability. They will change the format, when they need it, and they won't tell you.

> For the second point: It is sad that the website isn't always correct & up to date and that is an entirely different problem that should be tackled
How do you want to tackle this?

In the end, I will not object, if you really want to merge this, because I finished. All I can say is, that I wouldn't use the software, if I knew the data would be extracted with a fragile (no critic on your work, rather a general observation from the circumstances) script, which parses from HTML.
And making an error in the data might have serious consequences for a student.

**Pull request merged**